### DISCUSSION TITLE
数据源切换问题

### DISCUSSION THREAD

**Discussion by nizpxox** [zuihou/zuihou-admin-cloud#300] [02.提问]
不是动态切换数据源嘛，为什么会查默认表
![image](https://github.com/dromara/lamp-cloud/assets/141626156/604aa339-a149-47e2-ae43-d1ddd527b021)

**Comment by zuihou** [comment]
先按这个排查：  https://tangyh.top/doc/advanced/tenant/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98.html#%E6%89%A7%E8%A1%8Csql%E6%97%B6-%E4%B8%BA%E4%BB%80%E4%B9%88%E6%8A%A5-sqlsyntaxerrorexception-table-lamp-ds-c-defaults-base-parameter-doesn-t-exist

**Reply by nizpxox** [reply]
查完了代码没问题，我用租户的账号登录传租户的token也不行，查的还是def表

**Reply by nizpxox** [reply]
![image](https://github.com/dromara/lamp-cloud/assets/141626156/5bd36f5a-8551-449c-a5ad-f6ce988b0f43)

**Reply by nizpxox** [reply]
![image](https://github.com/dromara/lamp-cloud/assets/141626156/0f71f42d-7b79-485b-b0d7-7b53cb4d7f1d)

**Comment by zuihou** [comment]
没看到你的请求头

**Reply by nizpxox** [reply]
![image](https://github.com/dromara/lamp-cloud/assets/141626156/183e1098-ced9-490b-a436-4e823d8ef803)
这个嘛

**Reply by zuihou** [reply]
请求头中缺少 TenantId

**Reply by zuihou** [reply]
<img width="2265" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/177879a1-a119-4293-bb64-54a136c7afc8">

**Comment by zuihou** [comment]
<img width="1207" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/db22edac-5521-4e19-9df6-b62005d3a78b">

1.  断点调试，截图 tenant 的值

**Comment by zuihou** [comment]
<img width="1282" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/2c152856-2c3a-4fb1-a721-489d2e099f3d">

2. 断点调试， 截图： ContextUtil.getTenantId()

**Comment by zuihou** [comment]
![image](https://github.com/dromara/lamp-cloud/assets/8500410/154a5f2a-475c-4d8a-b0e7-83495fa3838d)

3.  访问base服务的 druid，截图有哪些数据源。 （端口号要以base服务的为准）
